src/mono/mono/mini/mini-wasm-debugger.c

```diff
@@ -1514,6 +1526,111 @@ describe_variable (InterpFrame *frame, MonoMethod *method, MonoMethodHeader *hea
 	return describe_value(type, addr, gpflags);
 }
 
+static gboolean
+decode_value (MonoType *t, guint8 *addr, const char* variableValue)
+{
+	char* endptr;
+	errno = 0;
+	switch (t->type) {
+	case MONO_TYPE_BOOLEAN:
+		if (!strcmp(variableValue, "True"))
+			*(guint8*)addr = 1;
+		else if (!strcmp(variableValue, "False"))
+			*(guint8*)addr = 0;
+		else 
+			return FALSE;
+		break;
+	case MONO_TYPE_CHAR:
+		*(gunichar2*)addr = variableValue[0];
+		break;
+	case MONO_TYPE_I1:
+		*(gint8*)addr = strtoimax(variableValue, &endptr, 10);
```

Comment:
Should this set only if there was no error?

Reply:
Yess, thanks!!!